### PR TITLE
Salto-3875: Improve fields system notes query that is used for chaged by/at

### PR DESCRIPTION
**Pull request by Shulik95**
_Simplified systemNotes field query to help avoid timeouts_

---

_None_

---
_Release Notes_: 
_None_

---
_User Notifications_: 
_None_

### CODE REVIEW

File: packages/netsuite-adapter/src/filters/author_information/system_note.ts [hunk 1]
```diff
@@ -91,10 +91,10 @@ const buildFieldSystemNotesQuery = (
   const whereQuery = fieldIds
     .map(toFieldWhereQuery)
     .join(' OR ')
-  return `SELECT name, field, recordid, date from (SELECT name, field, recordid, ${toSuiteQLSelectDateString('MAX(date)')} AS date`
-    + ` FROM (SELECT name, REGEXP_SUBSTR(field, '^(${FOLDER_FIELD_IDENTIFIER}|${FILE_FIELD_IDENTIFIER})')`
-    + ` AS field, recordid, date FROM systemnote WHERE ${toDateQuery(lastFetchTime)} AND (${whereQuery}))`
-    + ' GROUP BY name, field, recordid) ORDER BY name, field, recordid ASC'
+  return `SELECT name, field, recordid, ${toSuiteQLSelectDateString('date')} AS date`
+    + ` From systemnote WHERE ${toDateQuery(lastFetchTime)} AND (${whereQuery})`
```
Comment (dantal4) [inline packages/netsuite-adapter/src/filters/author_information/system_note.ts:95]: for consistency
```suggestion
    + ` FROM systemnote WHERE ${toDateQuery(lastFetchTime)} AND (${whereQuery})`
```

File: packages/netsuite-adapter/src/filters/author_information/system_note.ts [hunk 2]
```diff
@@ -91,10 +91,10 @@ const buildFieldSystemNotesQuery = (
...
+    + ` GROUP BY name, field, recordid, ${toSuiteQLSelectDateString('date')}`
```
Comment (dantal4) [inline packages/netsuite-adapter/src/filters/author_information/system_note.ts:96]: why do we need to group by the date? we want only the latest date for each (name, field, recordid) tuple.

**Review by dantal4 — COMMENTED** [review]
Nice work! Had a question about the grouping.

File: packages/netsuite-adapter/src/filters/author_information/system_note.ts (hunk 2 above)
Comment (Shulik95) [reply packages/netsuite-adapter/src/filters/author_information/system_note.ts:95]: We don't, It was a misunderstanding on my part. fixed it 👍

**Review by dantal4 — APPROVED** [review]

**Pull request merged**